src/govuk/helpers/typography.test.js

```diff
@@ -419,4 +413,145 @@ describe('@mixin govuk-typography-responsive', () => {
+        .foo {
+          font-family: "GDS Transport", arial, sans-serif;
+          -webkit-font-smoothing: antialiased;
+          -moz-osx-font-smoothing: grayscale;
+          font-weight: 400;
+          font-size: 12px;
+          font-size: 0.75rem;
+          line-height: 1.25; }
+          @media print {
+            .foo {
+              font-family: sans-serif; } }
+          @media (min-width: 30em) {
+            .foo {
+              font-size: 14px;
+              font-size: 0.875rem;
+              line-height: 1.42857; } }`)
+    })
+
+    it('enables tabular numbers opentype feature flags if $tabular: true', async () => {
+      const sass = `
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-font($size: 14, $tabular: true)
+      }`
+
+      const results = await renderSass({ data: sass, ...sassConfig })
+      const css = results.css.toString()
+
+      expect(css).toContain('font-feature-settings: "tnum" 1;')
+      expect(css).toContain(outdent`
+      ${outdent}
+        @supports (font-variant-numeric: tabular-nums) {
+          .foo {
+            font-feature-settings: normal;
+            font-variant-numeric: tabular-nums; } }`)
+    })
+
+    it('uses the tabular font instead if defined and $tabular: true', async () => {
+      const sass = `
+      $govuk-font-family-tabular: "ntatabularnumbers";
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-font($size: 14, $tabular: true)
+      }`
+
+      const results = await renderSass({ data: sass, ...sassConfig })
+      const css = results.css.toString()
+
+      expect(css).toContain('font-family: "ntatabularnumbers"')
+      expect(css).not.toContain('font-feature-settings')
+    })
+
+    it('sets font-size based on $size', async () => {
+      const sass = `
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-font($size: 12)
+      }`
+
+      const results = await renderSass({ data: sass, ...sassConfig })
+
+      expect(results.css.toString()).toContain('font-size: 12px')
+      expect(results.css.toString()).not.toContain('font-size: 14px')
+    })
+
+    it('does not output font-size if $size: false', async () => {
+      const sass = `
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-font($size: false)
+      }`
+
+      const results = await renderSass({ data: sass, ...sassConfig })
+
+      expect(results.css.toString()).not.toContain('font-size')
+    })
+
+    it('sets font-weight based on $weight', async () => {
+      const sass = `
+      // Avoid font face being output in tests
+      $govuk-include-default-font-face: false;
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-font($size: 14, $weight: bold)
+      }`
+
+      const results = await renderSass({ data: sass, ...sassConfig })
+
+      expect(results.css.toString()).toContain('font-weight: 700')
+    })
+
+    it('does not output font-weight if $weight: false', async () => {
```

Comment:
@36degrees do you think it's worth having a test for an unrecognised $weight?

Reply:
Hmm, good question… at the minute it'll basically ignore `$weight if it's not `regular` or `bold`. What would the assertion be for that test? Just check that the output does not contain `font-weight`?

Reply:
I think so, yep. If that's the behaviour we want to keep!

Reply:
Added one here – https://github.com/alphagov/govuk-frontend/pull/2474/files#diff-1aafb81d87bfc00119f85ee5e8f77ebdb31964d61786c89d2b51746be0cd943dR546-R559